src/Module/OStatus/Subscribe.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\OStatus;
+
+use Friendica\App;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig\Capability\IManagePersonalConfigValues;
+use Friendica\Core\Protocol;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\APContact;
+use Friendica\Model\Contact;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPClient\Capability\ICanSendHttpRequests;
+use Friendica\Network\HTTPClient\Client\HttpClientAccept;
+use Friendica\Protocol\ActivityPub;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class Subscribe extends \Friendica\BaseModule
+{
+	public function __construct(App\Page $page, IManagePersonalConfigValues $pConfig, ICanSendHttpRequests $httpClient, SystemMessages $systemMessages, IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->session        = $session;
+		$this->systemMessages = $systemMessages;
+		$this->httpClient     = $httpClient;
+		$this->pConfig        = $pConfig;
+		$this->page           = $page;
+	}
+
+	protected function content(array $request = []): string
+	{
+		if (!$this->session->getLocalUserId()) {
+			$this->systemMessages->addNotice($this->t('Permission denied.'));
+			$this->baseUrl->redirect('ostatus/subscribe');
```

Comment:
Isn't this a endless loop, is it?

Reply:
Good catch, it was present in the original module as well.